src/socket.c

```diff
@@ -341,7 +341,7 @@ static int connSocketIsLocal(connection *conn) {
     if (connSocketAddr(conn, cip, sizeof(cip) - 1, NULL, 1) == C_ERR)
         return -1;
 
-    return !strcmp(cip,"127.0.0.1") || !strcmp(cip,"::1");
+    return !strncmp(cip, "127.", 4) || !strncmp(cip, "::1", 3);
```

Comment:
For IPv4, looks good.

For IPv6, this check looks for cip starting with "::1". That would also match "::1234:6789:abcd..." which is not a local IP. I think the old strcmp is better.

Reply:
Sorry I missed this message. Thanks for your review. Code is updated.